app/script_variant.py

```diff
@@ -20,7 +20,14 @@ def read_dictionary(file_name):
             with open(file_name, 'r') as f:
                 for line in f:
                     kanji, hiragana = line[:-1].split('\t')
```

Comment:
Use .rstrip('\n') instead of [:-1], which removes trailing '\n'. It's the same effect, but the intent is clearer.